src/app/(maps)/components/FullScreenMap.tsx

```diff
@@ -32,3 +53,47 @@ const getVisitorLocation = async (): Promise<{ longitude: number, latitude: numb
     return undefined
   }
 }
+
+function useCameraParams (): { toUrl: (camera: CameraInfo) => string, fromUrl: () => CameraInfo | null } {
+  const pathname = usePathname()
+  const initialSearchParams = useSearchParams()
+
+  function toUrl (camera: CameraInfo): string {
+    const params = new URLSearchParams(initialSearchParams)
+    params.delete('camera')
+    params.append('camera', cameraInfoToQuery(camera))
+
+    return `${pathname}?${params.toString()}`
+  }
+
+  function fromUrl (): CameraInfo | null {
+    const cameraParams = initialSearchParams.get('camera')
+    if (cameraParams == null) {
+      return null
+    }
+
+    return queryToCameraInfo(cameraParams)
+  }
+
+  return { toUrl, fromUrl }
+}
+
+const cameraInfoToQuery = ({ zoom, center }: CameraInfo): string => {
+  return `${Math.ceil(zoom)}/${center.lat.toPrecision(3)}/${center.lng.toPrecision(3)}`
```

Comment:
- For latitude/longitude can you preserve at least 5 decimal places? 
 https://en.wikipedia.org/wiki/Decimal_degrees#Precision
- is it possible to keep the `/` in the url?

Reply:
1. Done
2. It is not possible to do that using URLSearchParams, as it automaticaly encodes special characters.
I probably can prevent that by manually construction the parameter. However, it takes off any benefits that URLSearchParams gives.

Therefore, I believe it will be more beneficial to use URLSearchParams and put up with percent encoded (%2f) symbols in the url.

But if you're ok with keeping '/' slash in the url, you can check out the changes that I made